src/libsync/theme.cpp

```diff
@@ -106,6 +106,16 @@ QString Theme::appName() const
     return APPLICATION_SHORTNAME;
 }
 
+QUrl Theme::stateOnlineImageSource() const
+{
+    return QUrl("qrc" + themeImagePath("state-ok"));
```

Comment:
I think I'd be a bit more conservative there and check if the path starts with ':' or not as to not make too many assumptions about what themeImagePath returns. Because if it wouldn't start with ':' we'd want to use `QUrl::fromLocalFile()` instead.

So I think I'd got for something more verbose like this:
```
const auto imagePath = themeImagePath("state-ok");
if (imagePath.startsWith(':')) {
    auto url = QUrl();
    url.setScheme(QStringLiteral("qrc"));
    url.setPath(imagePath.mid(1));
    return url;
} else {
    return QUrl::fromLocalFile(imagePath);
}
```

To avoid code duplication you might want to get the if part factored out in a function in the anonymous namespace or such.

PS: In my opinion all that work should be done inside `QUrl::fromLocalFile()` but let's not get into patching QUrl just now. ;-)